### ISSUE TITLE
Do I need the NVIDIA CUDA Toolkit along with conda installation of the CUDA toolkit?

### ISSUE THREAD

**Issue by drag05** [tensorflow/tensorflow#33695]
Version mismatch issues encountered at the installation of Tensorflow with local GPU support led me **question the need for the coexistence on the same machine of both CUDA packages**, namely: 

The NVIDIA **CUDA Toolkit** along with **CUDNN**

<img width="257" alt="Annotation 2019-10-24 122829" src="https://user-images.githubusercontent.com/39677929/67505904-1ae1b580-f65a-11e9-9c18-78c22b4266e2.png">

and the `conda` installation of the **cudatoolkit** along with **cudnn** 

<img width="311" alt="Annotation 2019-10-24 121845" src="https://user-images.githubusercontent.com/39677929/67506044-5c726080-f65a-11e9-8f24-d1286d9da45d.png">

As shown above, on my machine these packages have different versions but tensorflow is configured for GPU support and is working:

`sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))`

outputs something along the lines:

<img width="401" alt="Annotation 2019-10-24 124233" src="https://user-images.githubusercontent.com/39677929/67507285-99d7ed80-f65c-11e9-8de5-ba3fa440b3b4.png">

Conda installation of tensorflow-gpu via 

` conda create -n tensorflow tensorflow-gpu`

seems to resolve version mismatch issues encountered with `pip install`, `keras::install_keras()` or with `tensorflow::install_tensorflow()`

The reason for my question is twofold: 

1. Beside CUDA,  the NVIDIA CUDA Toolkit installs a plethora of tools that may not be used by everyone therefore, a guide to the custom installation of the CUDA Toolkit would be more than welcome;

2. The _cudart64-100.dll cannot be found_ error popping at version mismatch between CUDA/CUDNN, Python and Tensorflow  refers to the _dll_ library located in the `conda` environment and not to the NVIDIA CUDA installation (which in my case contains the _cudart64-101.dll_ and still works)

**Comment by JulianOrteil** [comment]
No, you don't need both; there are two probable issues you are encountering:

1st: I don't believe TF supports CUDA 10.1 yet (_**I could be wrong**_). I know for my build I use CUDA 10.0 with cuDNN v7.50. You can see in your conda installation you have CUDA 10.0; not 10.1.

Download v10.0: https://developer.nvidia.com/cuda-10.0-download-archive?target_os=Windows&target_arch=x86_64

2nd: Check to ensure you have the following paths in your environment variables **and** above conda's paths:
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\\<VERSION_HERE>\bin
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\\<VERSION_HERE>\extras\CUPTI\libx64
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\\<VERSION_HERE>\include
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\\<VERSION_HERE>\lib\x64

EXTRA: When I install CUDA, I specifically uncheck the following settings (through advanced setup):
1. Documentation (in CUDA)
2. Visual Studio Integration (unless you will be using Visual Studio [not VS Code]) (in CUDA)
3. Samples (in CUDA)
4. Driver components

It keeps the size down and speeds up installation. You do not have to do this if you don't want to.

After CUDA is installed, install the latest graphics driver for your graphics card from NVIDIA. Everything unchecked in Driver components is installed when you install the graphics driver.

**Comment by drag05** [comment]
@JulianOrteil

 Thank you for your time! 

I am a bit confused by your answer as my question referred to the necessity of co-existence of both NVIDIA CUDA and conda `cudatoolkit` installations.

Since your answer to this question is no, am I correct to uninstall Nvidia CUDA/CUDNN and leave the conda `cudatoolkit` installation ONLY?

If it is correct to remove NVIDIA CUDA/CUDNN installation, then should all environment variables pointing to the NVIDIA CUDA installation be removed?

I am asking all these questions because tensorflow seems to be looking for `cudatoolkit` and not for NVIDIA CUDA install and I am supporting this observation with the fact that every time I installed mismatched versions of tensorflow/python I got the error "_cudart64-100.dll cannot be found_" - event which coincided with  `cudatoolkit` missing from respective conda environment.

This mismatch was automatically solved by a conda installation such as:

 'conda create -tensorflow tensorflow-gpu` 

which took care automatically of tensorflow-gpu/python version matching and also automatically installed the `cudatoolkit` v.10.0 to which tensorflow points. 

NVIDIA CUDA installation doesn't seem to exist although is installed and respective environment variables are set!

So, in summary: will tensorflow-gpu work after uninstalling NVIDIA CUDA and removing all pertaining environment variables leaving only conda installation? 

Thank you!

**Comment by JulianOrteil** [comment]
@drag05 

Sorry for the confusion! I installed TF directly into Python installed in my user not a conda environment.

If you installed Tensorflow using a conda environment, then no you should not need the NVIDIA CUDA/cuDNN files. You can also remove them from path, in fact you should to keep your path size down.

Just make sure you do not remove any Anaconda paths.

As to your last question, I am unsure if you need those. If you would like to experiment with that and it works without them, let me know! (I would rather not experiment in my environment since I am training models at the current moment)

One last thing, even though you do not need the CUDA/cuDNN files, you do need to ensure your graphics card driver is above 390 (I believe it is that version). Just install the latest and you should be fine.

~Jules

**Comment by drag05** [comment]
@JulianOrteil 

Thank you for the prompt response! 
I have uninstalled Nvidia CUDA/CUDNN and removed all pertaining environment variables. 

Tested tensorflow-gpu/keras on RStudio and still works. 

Also tested the gaming and still woks. 

I guess tensorflow-gpu installation instructions would be less confusing if were split into the option of installing from Python OR the option of installing from *conda. 

Cheers!
